test/serializer_test.dart

```diff
@@ -305,21 +318,72 @@ main() {
+      expect(json, '{"@dart_type":"Complex","nums":[1,2.2,3],"strings":["1","2","3"],"bools":[true,false,true],"ints":[1,2,3],"doubles":[1.1,2.2,3.3],"dates":["2016-12-24T00:00:00.000","2016-12-25T00:00:00.000","2016-12-26T00:00:00.000"],"ignores":[{"@dart_type":"WithIgnore","a":"1337A","b":"42A"},{"@dart_type":"WithIgnore","a":"1337B","b":"42B"}],"numSet":{"numA":1,"numB":12.2},"stringSet":{"strA":"1","strB":"3"},"boolSet":{"ok":true,"nok":false},"intSet":{"intA":1,"intB":12},"doubleSet":{"dblA":1,"dblB":12},"ignoreSet":{"A":{"@dart_type":"WithIgnore","a":"1337A","b":"42A"},"B":{"@dart_type":"WithIgnore","a":"1337B","b":"42B"}}}');
+    });
+
+    test("Deserialize", () {
+      Complex complex = Serializer.fromJson('{"@dart_type":"Complex","nums":[1,2.2,3],"strings":["1","2","3"],"bools":[true,false,true],"ints":[1,2,3],"doubles":[1.1,2.2,3.3],"dates":["2016-12-24T00:00:00.000","2016-12-25T00:00:00.000","2016-12-26T00:00:00.000"],"ignores":[{"@dart_type":"WithIgnore","a":"1337A","b":"42A"},{"@dart_type":"WithIgnore","a":"1337B","b":"42B"}],"numSet":{"numA":1,"numB":12.2},"stringSet":{"strA":"1","strB":"3"},"boolSet":{"ok":true,"nok":false},"intSet":{"intA":1,"intB":12},"doubleSet":{"dblA":1,"dblB":12},"dateSet":{"fiesta":"2016-12-24T00:00:00.000","christmas":"2016-12-25T00:00:00.000"},"ignoreSet":{"A":{"@dart_type":"WithIgnore","a":"1337A","b":"42A"},"B":{"@dart_type":"WithIgnore","a":"1337B","b":"42B"}}}');
+
+      expect(complex.nums,    [ 1, 2.2, 3 ]);
+      expect(complex.strings, [ "1", "2", "3" ]);
+      expect(complex.bools,   [ true, false, true ]);
+      expect(complex.ints,    [ 1, 2, 3 ]);
+      expect(complex.doubles, [ 1.1, 2.2, 3.3 ]);
+      expect(complex.dates,   [ new DateTime(2016,12,24), new DateTime(2016,12,25), new DateTime(2016,12,26)]);
+      expect(complex.ignores[0].a,      "1337A");
+      expect(complex.ignores[0].b,      "42A");
+      expect(complex.ignores[0].secret, null);
+      expect(complex.ignores[1].a,      "1337B");
+      expect(complex.ignores[1].b,      "42B");
+      expect(complex.ignores[1].secret, null);
+
+      expect(complex.numSet   , { "numA": 1, "numB": 12.2 });
+      expect(complex.stringSet, { "strA": "1", "strB": "3" });
+      expect(complex.boolSet  , { "ok": true, "nok": false });
+      expect(complex.intSet   , { "intA": 1, "intB": 12 });
+      expect(complex.doubleSet, { "dblA": 1, "dblB": 12 });
+//      expect(complex.dateSet  , { "fiesta": new DateTime(2016,12,24), "christmas": new DateTime(2016,12,25) });
```

Comment:
I think the comparison beetween 2 dates do not work in the `except`, did you try to compare with the `toIso8601String()` method
```
expect(complex.dateSet  , { "fiesta": new DateTime(2016,12,24).toIso8601String(), "christmas": new DateTime(2016,12,25).toIso8601String() });
```

to be sure that it's really a Datetime, you can test the type of `complex.dataset["fiesta"]`